### DISCUSSION TITLE
Suggested features :-)

### DISCUSSION THREAD

**Discussion by ghost** [larsonmars/sysvars#1] [Ideas]
HI,
first - thank you for this nice tool! I got few ideas, you could add meanwhile:
- support for bsdsocket.library (to check if inet connection is active);
- device query - for example, check if there is a something connected to [name].device, unit [x];
- PPC detection?

**Comment by larsonmars** [comment]
Thank you for your suggestions!

I think detection of bsdsocket.library (in the sense that it - the library - is available) should be trivial. This seems like a valuable info having as environment variable. Whether an INET connection is active or not is rather volatile. You would have to call the tool continuously (like a service) to have the env.variable updated. Or add service capability to the tool directly, which is a whole different level. Could you elaborate on the use case that you would like to solve with this?

Regarding specific device query, there are tools available, such as in the [boards.library](https://aminet.net/package/util/libs/BoardsLib) and [identify.library](https://github.com/shred/identify) that might do what you want. If not, could you elaborate how you would like to use sysvars for this?

PPC detection could be a bit dificult for me to implement, because I have no such equipment and no experience with that sort of thing. However, I think WinUAE comes with support for PPC expansion boards. So I will consider this in one of the future updates. Perhaps it is just to query the manufacturer/device IDs associated with a PPC. Anyway, this is exactly where the community here could step in to give some hints, or even contribute some code :relaxed:

**Reply by ghost** [reply]
Thanks for quick reply! Here is my detailed explanation:

1. bsdsocket.libbrary - I'd just like to know if the Internet stack is running. Now I got a script which fires it up and set "internet" variable in ENV manually - then I use that variable to display information about active internet connection on WB title-bar. That's it.
2. Maybe there is an easy way in AmigaOS to query device/unit and get in response that something (whatever!) is available/connected? I don't know :-) But it would be cool to have a quick way to check if something is connected to the Amiga.

**Reply by larsonmars** [reply]
Ok, so bsdsocket.library detection has been added to v0.10

Regarding the expansions, I am discussing this with Marcus Gerards, the maintainer of the boards.library. I still do not really know what would be the best way to put the information into variables. Currently, if something is not detected, its variable is set without content. Clearly, we do not want hundreds of empty environment variables. Thus, some kind of clustering makes more sense, or not defining a variable instead of defining it without content...

**Reply by bubbob42** [reply]
@arti040  You can use ShowBoards from the boards.library package to check for specific expansions. The library is shipped with OS3.2+, but ShowBoards itself is part of the Aminet package only:

https://aminet.net/util/libs/BoardsLib.lha

Checking the internet could require different methods depending on the stack. Some ship bsdsocket, some generate it on the fly (Miami), some have stack-specific tools to check whether the machine has a connection.

**Reply by larsonmars** [reply]
@arti040: Are your issues covered? Can we close this discussion? Could you create issues for the missing features (e.g., the PPC detection)?
